vscode-lean4/src/taskgutter.ts

```diff
@@ -51,30 +66,57 @@ class LeanFileTaskGutter {
 }
 
 export class LeanTaskGutter implements Disposable {
-    private decoration: TextEditorDecorationType;
-    private status: { [uri: string]: number | undefined } = {};
+    private decorations: Map<LeanFileProgressKind, [TextEditorDecorationType, string]> = new Map<LeanFileProgressKind, [TextEditorDecorationType, string]>();
+    private status: { [uri: string]: Map<LeanFileProgressKind, number> | undefined } = {};
     private gutters: { [uri: string]: LeanFileTaskGutter | undefined } = {};
     private subscriptions: Disposable[] = [];
 
     constructor(client: LeanClient, context: ExtensionContext) {
-        this.decoration = window.createTextEditorDecorationType({
-            overviewRulerLane: OverviewRulerLane.Left,
-            overviewRulerColor: 'rgba(255, 165, 0, 0.5)',
-            dark: {
-                gutterIconPath: context.asAbsolutePath('media/progress-dark.svg'),
-            },
-            light: {
-                gutterIconPath: context.asAbsolutePath('media/progress-light.svg'),
-            },
-            gutterIconSize: 'contain',
-        });
+        this.decorations.set(LeanFileProgressKind.Processing, [
```

Comment:
Is this where you get the exception?  And it goes away if you replace LeanFileProgressKind.Processing by 1?

Reply:
Yes, here and everywhere else in this file where I write `LeanFileProgressKind.something`.

Reply:
Apparently `self` is a browser-only feature and does not exist on node.js: https://developer.mozilla.org/en-US/docs/Web/API/Window/self

The take-away is that we probably shouldn't import `@lean4/infoview` in the extension host at all (since it depends on browser code which may fail when run outside of a browser).  For now, please just use 1 instead of the constant.

@Vtec234 Do you have any ideas on how to fix this properly?  Should we make a third package that contains the platform-independent code?  This was all a bit easier before the lerna setup, when we could just (and did) import single files.

Reply:
Ahh that is annoying, afaik the usual solution is for a package to have two entry points, one for the browser and one elsewhere. NPM provides ways to set this up in `package.json`. Please feel free to try, otherwise I can look into it later next week.